pkg/generate_test.go

```diff
@@ -846,14 +1157,71 @@ func TestGenerateOverlays(t *testing.T) {
 					t.Errorf("kustomize file does not exist at path %v", kustomizationFilepath)
 				}
 
+				if tt.options.IsKubernetesCluster {
+					// Validate that the ingress.yaml got created successfully and contains the proper entries
+					ingressFilepath := filepath.Join(tt.outputFolder, ingressFileName)
+					exists, err = tt.fs.Exists(ingressFilepath)
+					if err != nil {
+						t.Errorf("unexpected error checking if ingress file exists %v", err)
+					}
+					if !exists && tt.options.TargetPort != 0 {
+						t.Errorf("ingress file does not exist at path %v", ingressFilepath)
+					} else if exists {
+						ingress := networkingv1.Ingress{}
+						ingressBytes, err := tt.fs.ReadFile(ingressFilepath)
+						if err != nil {
+							t.Errorf("unexpected error reading deployment file")
+						}
+						err = yaml.Unmarshal(ingressBytes, &ingress)
+						if err != nil {
+							t.Errorf("unexpected error unmarshaling file")
+						}
+						if len(tt.options.KubernetesResources.Ingresses) > 0 && ingress.Name != tt.options.KubernetesResources.Ingresses[0].Name {
+							t.Errorf("expected ingress name %v, got %v", tt.options.KubernetesResources.Ingresses[0].Name, ingress.Name)
+						} else if len(tt.options.KubernetesResources.Ingresses) == 0 && ingress.Name != tt.options.Name {
+							t.Errorf("expected ingress name %v, got %v", tt.options.Name, ingress.Name)
+						}
+					}
+
+				}
+
+				if !tt.options.IsKubernetesCluster {
+					// Validate that the route.yaml got created successfully and contains the proper entries
+					routeFilepath := filepath.Join(tt.outputFolder, routeFileName)
+					exists, err = tt.fs.Exists(routeFilepath)
+					if err != nil {
+						t.Errorf("unexpected error checking if ingress file exists %v", err)
```

Comment:
I think it's route file instead of ingress

Reply:
updated